elmslie-core/src/main/java/vivid/money/elmslie/core/store/ElmStore.kt

```diff
@@ -27,18 +28,23 @@ class ElmStore<Event : Any, State : Any, Effect : Any, Command : Any>(
     private val effectsBuffer = EffectsBuffer(effectsInternal)
     private val eventsInternal = PublishSubject.create<Event>()
     private val commandsInternal = PublishSubject.create<Command>()
+    private val isStartedInternal = AtomicBoolean(false)
 
     override val effects: Observable<Effect> = effectsBuffer.getBufferedObservable()
     override val states: Observable<State> = statesInternal.distinctUntilChanged()
     override val currentState: State get() = statesInternal.value!!
-    @Volatile
-    override var isStarted = false
+
+    override val isStarted
+        get() = isStartedInternal.get()
 
     override fun accept(event: Event) = eventsInternal.onNext(event)
 
     override fun start(): Store<Event, Effect, State> {
-        if (isStarted) {
-            logger.fatal("Store is already started. Usually, it happened inside StoreHolder.")
+        if (!isStartedInternal.compareAndSet(false, true)) {
```

Comment:
Maybe it's too much to use AtomicBoolean if we can guarantee that start will be always called from the same thread. What do you think?

Reply:
It's also used inside addChildStore in Disposable#isDisposed method, which may be called from a background thread, so i guess it's better to leave it as is

Reply:
Btw, it's a little bit confusing that child store starts on attach. Why so? Maybe save children as a list inside parent and start/stop all of them with parent store?

Reply:
It's an old issue, i agree that it's better to ensure that parent and child store lifecycles are in sync. 

It's more clear to me now that isStarted will be called from the main thread in standard cases, but it's still a public property of Store. I suppose it's better to keep it as Atomic since the rest of Store methods are thread safe

Reply:
@diklimchuk maybe we can add such types of improvements as issues inside github? Just to be sure that we will not forget about it.

Reply:
@opengamer29 I keep a separate backlog for myself in trello, but i suppose it would be better to start using issues as soon as someone else will start looking at them. Do you have plans to contribute to the library and look at those issues?

Reply:
@diklimchuk I'm sure that I will not have a lot of time to do it, but I can do something from time to time or if it is needed by the main project. I assume that other guys from the team will be interested in this backlog too.

Reply:
I've created an [issue](https://github.com/vivid-money/elmslie/issues/47) to fix thix